### PR TITLE
fix: disable application database configuration in replica mode

### PR DESCRIPTION
**Pull request by litaocdl**
this patch add web hook to not configure the application database 
if a cluster is created as a replica cluster. 

Close: #253 
Signed-off-by: Tao Li <tao.li@enterprisedb.com>

### CODE REVIEW

**Review by jlong49 — APPROVED** [review]

**Review by hh24k — APPROVED** [review]

**Comment by litaocdl** [comment]
resolve the conflict again

**Comment by litaocdl** [comment]
all test pass

**Comment by litaocdl** [comment]
rebase the pr with latest change.

**Review by gbartolini — APPROVED** [review]

File: docs/src/bootstrap.md [hunk 1]
```diff
@@ -528,15 +528,16 @@ spec:
 
 #### Configure the application database
 
-For the cluster recovered, we can configure the application database name and credentials with 
-additional configuration. To update application database credentials, we can generate our own 
-passwords, store them as secrets, and update the database use the secrets. Or we can also let the 
-operator generate a secret with randomly secure password for use. Please reference the 
-["Bootstrap an empty cluster"](#bootstrap-an-empty-cluster-initdb) section for more information 
-about secrets.
+For the recoverd cluster, we can configure the application database name and
```
Comment (drsm79) [inline docs/src/bootstrap.md:531]: s/recoverd/recovered/

File: docs/src/bootstrap.md [hunk 2]
```diff
@@ -528,15 +528,16 @@ spec:
...
+credentials with additional configuration. To update application database
+credentials, we can generate our own passwords, store them as secrets, and
+update the database use the secrets. Or we can also let the operator generate a
+secret with randomly secure password for use. Please reference the
+["Bootstrap an empty cluster"](#bootstrap-an-empty-cluster-initdb)
+section for more information about secrets.
 
-The following example configure the application database `app` with owner `app`,  and supplied secret
-`app-secret`.
+The following example configure the application database `app` with owner
```
Comment (drsm79) [inline docs/src/bootstrap.md:539]: Are whitespace rules documented & enforced somewhere?

File: docs/src/bootstrap.md [hunk 3]
```diff
@@ -552,7 +553,7 @@ spec:
       [...]
 ```
 
-With the above configuration, following will happen after recovery
+With the above configuration, the following will happen after recovery is completed:
```
Comment (drsm79) [inline docs/src/bootstrap.md:556]: s/is completed/completes/

File: docs/src/bootstrap.md [hunk 4]
```diff
@@ -562,8 +563,9 @@ as owner of database `app`.
 application database will be changed to the value of `password` in secret. 
 
 !!! Important
-    Application database configuration will be disabled if the new cluster is 
-    created as a replica cluster, with replica mode enabled. 
+    In case of a replica cluster with replica mode enabled, the operator
```
Comment (drsm79) [inline docs/src/bootstrap.md:566]: How about:

For a replica cluster with replica mode enabled, the operator will not create any database or user in the PostgreSQL instance, as these will be recovered from the original cluster.

?

File: docs/src/bootstrap.md [hunk 5]
```diff
@@ -806,8 +808,9 @@ With the above configuration, following will happen after bootstrap
    application database will be changed to the value of `password` in secret.
 
 !!! Important
-    Application database configuration will be disabled if the new cluster is
-    created as a replica cluster, with replica mode enabled.
+    In case of a replica cluster with replica mode enabled, the operator
```
Comment (drsm79) [inline docs/src/bootstrap.md:811]: Should match above, if that change is done.

File: docs/src/bootstrap.md (hunk 3 above)
Comment (gbartolini) [reply docs/src/bootstrap.md:556]: I had missed the second fix. Thanks.

File: docs/src/bootstrap.md (hunk 4 above)
Comment (gbartolini) [reply docs/src/bootstrap.md:566]: Sounds good.

**Review by mnencia — APPROVED** [review]

**Pull request merged**